vortex-array/src/array/constant/compute.rs

```diff
@@ -99,30 +95,17 @@ impl SearchSortedFn for ConstantArray {
     }
 }
 
-impl CompareFn for ConstantArray {
-    fn compare(&self, rhs: &Array, operator: Operator) -> VortexResult<Array> {
-        if let Some(true) = rhs.statistics().get_as::<bool>(Stat::IsConstant) {
-            let lhs = self.scalar();
-            let rhs = scalar_at(rhs, 0)?;
-
-            let scalar = scalar_cmp(lhs, &rhs, operator);
-
-            Ok(ConstantArray::new(scalar, self.len()).into_array())
+impl MaybeCompareFn for ConstantArray {
+    fn maybe_compare(&self, array: &Array, operator: Operator) -> Option<VortexResult<Array>> {
+        if let Some(true) = array.statistics().get_as::<bool>(Stat::IsConstant) {
+            Some({
+                let lhs = self.scalar();
+                let rhs = scalar_at(array, 0).expect("Expected scalar");
+                let scalar = scalar_cmp(lhs, &rhs, operator);
+                Ok(ConstantArray::new(scalar, self.len()).into_array())
+            })
         } else {
-            let datum = Arc::<dyn Datum>::from(self.scalar());
-            let rhs = rhs.clone().into_canonical()?.into_arrow();
-            let rhs = rhs.as_ref();
-
-            let boolean_array = match operator {
-                Operator::Eq => cmp::eq(datum.as_ref(), &rhs)?,
-                Operator::NotEq => cmp::neq(datum.as_ref(), &rhs)?,
-                Operator::Gt => cmp::gt(datum.as_ref(), &rhs)?,
-                Operator::Gte => cmp::gt_eq(datum.as_ref(), &rhs)?,
-                Operator::Lt => cmp::lt(datum.as_ref(), &rhs)?,
-                Operator::Lte => cmp::lt_eq(datum.as_ref(), &rhs)?,
-            };
-
-            Ok(Array::from_arrow(&boolean_array, true))
+            None
```

Comment:
We could centralize the Arrow Array<->Scalar compute here since the entry function will swap the arguments for us.